mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -1,59 +1,75 @@
-use zcash_primitives::block::BlockHash;
 use zcash_primitives::consensus::BlockHeight;
 use zcash_primitives::transaction::TxId;
-use zcash_primitives::zip32::ExtendedFullViewingKey;
 
-mod z_coin_grpc {
-    tonic::include_proto!("cash.z.wallet.sdk.rpc");
-}
-use crate::{RpcCommonOps, ZTransaction};
-use rpc::v1::types::H256 as H256Json;
-use z_coin_grpc::compact_tx_streamer_client::CompactTxStreamerClient;
-use z_coin_grpc::{BlockId, BlockRange, ChainSpec, CompactBlock as TonicCompactBlock,
+cfg_native!(
+    use crate::{RpcCommonOps, ZTransaction};
+    use super::CheckPointBlockInfo;
+
+    use crate::utxo::rpc_clients::{UtxoRpcClientOps, NO_TX_ERROR_CODE};
+    use db_common::sqlite::rusqlite::{params, Connection, Error as SqliteError, NO_PARAMS};
+    use db_common::sqlite::{query_single_row, run_optimization_pragmas};
+    use common::async_blocking;
+    use common::executor::Timer;
+    use common::log::{debug, error, info, LogOnError};
+    use common::Future01CompatExt;
+    use futures::channel::mpsc::channel;
+    use group::GroupEncoding;
+    use http::Uri;
+    use prost::Message;
+    use protobuf::Message as ProtobufMessage;
+    use std::path::{Path, PathBuf};
+    use std::pin::Pin;
+    use std::str::FromStr;
+    use std::time::Duration;
+    use tokio::task::block_in_place;
+    use tonic::transport::{Channel, ClientTlsConfig};
+    use zcash_client_backend::data_api::{BlockSource, WalletRead, WalletWrite};
+    use zcash_client_backend::data_api::chain::{scan_cached_blocks, validate_chain};
+    use zcash_client_backend::data_api::error::Error as ChainError;
+    use zcash_client_backend::proto::compact_formats::CompactBlock;
+    use zcash_primitives::block::BlockHash;
+    use zcash_primitives::zip32::ExtendedFullViewingKey;
+    use zcash_client_sqlite::error::SqliteClientError as ZcashClientError;
+    use zcash_client_sqlite::wallet::init::{init_accounts_table, init_blocks_table, init_wallet_db};
+    use zcash_client_sqlite::WalletDb;
+    use zcash_client_sqlite::error::SqliteClientError;
+
+    mod z_coin_grpc {
+        tonic::include_proto!("cash.z.wallet.sdk.rpc");
+    }
+
+    use rpc::v1::types::H256 as H256Json;
+    use z_coin_grpc::compact_tx_streamer_client::CompactTxStreamerClient;
+    use z_coin_grpc::{BlockId, BlockRange, ChainSpec, CompactBlock as TonicCompactBlock,
                   CompactOutput as TonicCompactOutput, CompactSpend as TonicCompactSpend, CompactTx as TonicCompactTx,
                   TxFilter};
+);
 
-pub type WalletDbShared = Arc<Mutex<WalletDb<ZcoinConsensusParams>>>;
-
+#[allow(unused)]
 struct CompactBlockRow {
     height: BlockHeight,
     data: Vec<u8>,
 }
 
+#[cfg(not(target_arch = "wasm32"))]
 pub type OnCompactBlockFn<'a> = dyn FnMut(TonicCompactBlock) -> Result<(), MmError<UpdateBlocksCacheErr>> + Send + 'a;
 
+#[cfg(target_arch = "wasm32")]
+#[allow(unused)]
+pub type OnCompactBlockFn<'a> = dyn FnMut(String) -> Result<(), MmError<UpdateBlocksCacheErr>> + Send + 'a;
```

Comment:
Here too, clippy didnt catch err for wasm target  without #[allow(unused)]

Reply:
same as above